packages/dds/tree/src/core/rebase/types.ts

```diff
@@ -108,6 +108,41 @@ export interface GraphCommit<TChange> {
 	inverse?: TChange;
 }
 
+/**
+ * The type of commit.
+ *
+ * @public
+ */
+export enum CommitKind {
+	/** A typical local commit */
+	Default,
+	/** A commit that is the result of an undo. */
+	Undo,
+	/** A commit that is the result of a redo. */
+	Redo,
+	/**
+	 * A commit that is the result of a rebase and should replace a previously generated revertible.
+	 * todo: improve error reporting in this case
+	 */
+	Rebase,
```

Comment:
We actually don't emit it at all and I'm not sure why it's here since updating revertibles is done automatically. I'll remove it for now and make sure Yann didn't add it for some special reason.